vault/resource_pki_secret_backend_role.go

```diff
@@ -512,6 +519,12 @@ func pkiSecretBackendRoleCreate(ctx context.Context, d *schema.ResourceData, met
 		data[consts.FieldPolicyIdentifiers] = pki.ReadPolicyIdentifierBlocks(policyIdentifierBlocksRaw.(*schema.Set))
 	}
 
+	if provider.IsAPISupported(meta, provider.VaultVersion110) {
+		if notAfter, ok := d.GetOk(consts.FieldNotAfter); ok {
+			data[consts.FieldNotAfter] = notAfter
```

Comment:
This field could just be added to the list of `pkiSecretFields` above as this version of the provider only supports Vault 1.10 and above. See https://registry.terraform.io/providers/hashicorp/vault/latest/docs/guides/version_4_upgrade#what-vault-server-versions-are-supported-in-version-4x